zebra-state/src/service/read/tree.rs

```diff
@@ -74,6 +74,19 @@ where
 {
     let mut db_list = db.sapling_subtree_list_by_index_for_rpc(start_index, limit);
 
+    if let Some(limit) = limit {
+        let subtrees_num = u16::try_from(db_list.len())
+            .expect("There can't be more than `u16::MAX` Sapling subtrees.");
+
+        // Return the subtrees if the amount of them reached the given limit.
+        if subtrees_num == limit.0 {
```

Comment:
Could we update the implementation of (`sapling_subtrees_in_range`](https://github.com/ZcashFoundation/zebra/blob/main/zebra-state/src/service/non_finalized_state/chain.rs#L710) to take a range?

(And update the docs for `sapling_subtree_list_by_index_for_rpc`)

Reply:
This change is trickier than it looks, because turning `start, len` into `start..(start + len)` could cause integer overflows or off-by-one errors.